cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,341 @@
+	for _, event := range filteredEvent {
+		if raw {
+			if event.CloudTrailEvent != nil {
+				fmt.Printf("%v \n\n", *event.CloudTrailEvent)
+			}
+		} else {
+			rawEventDetails, err := extractUserDetails(event.CloudTrailEvent)
+			if err != nil {
+				fmt.Printf("[Error] Error extracting event details: %v", err)
+			}
+			sessionIssuer := rawEventDetails.UserIdentity.SessionContext.SessionIssuer.UserName
+			if event.EventName != nil {
+				fmt.Printf("%v ", *event.EventName)
+			} else {
+				continue
+			}
+
+			if event.EventTime != nil {
+				fmt.Printf("|%v ", event.EventTime.String())
+				if event.Username != nil {
+					fmt.Printf("| User: %v ", *event.Username)
+				}
+				if sessionIssuer != "" {
+					fmt.Printf("| ARN: %v\n", sessionIssuer)
+				} else {
+					fmt.Print("\n")
+				}
+
+				if printUrl && event.CloudTrailEvent != nil {
+					if err == nil {
+						fmt.Printf("EventLink: %v\n\n", generateLink(*rawEventDetails))
+					} else {
+						fmt.Println("EventLink: <not available>")
+					}
+				}
+
+			}
+		}
+	}
+
+}
+
+func (o *LookupEventsOptions) run() error {
+
+	err := utils.IsValidClusterKey(o.clusterID)
+	if err != nil {
+		return err
+	}
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return fmt.Errorf("unable to create connection to ocm: %w", err)
+	}
+	defer connection.Close()
+
+	cluster, err := utils.GetClusterAnyStatus(connection, o.clusterID)
+	if err != nil {
+		return err
+	}
+	if strings.ToUpper(cluster.CloudProvider().ID()) != "AWS" {
+		return fmt.Errorf("[ERROR] this command is only available for AWS clusters")
+	}
+
+	Ignore, err := envConfig.LoadCloudTrailConfig()
+	if err != nil {
+		return fmt.Errorf("[ERROR] error Loading cloudtrail configuration file: %w", err)
+	}
+	if len(Ignore) == 0 {
+		fmt.Println("\n[WARNING] No filter list DETECTED!!")
+	}
+
+	cfg, err := osdCloud.CreateAWSV2Config(connection, cluster)
+	if err != nil {
+		return err
+	}
+	DefaultRegion := "us-east-1"
+	configOptions := config.WithRegion(DefaultRegion)
+	defaultConfig, err := config.LoadDefaultConfig(context.Background(), configOptions)
+	if err != nil {
+		return err
+	}
+
+	startTime, err := parseDurationToUTC(o.startTime)
+	if err != nil {
+		return err
+	}
+
+	fetchFilterPrintEvents := func(client cloudtrail.Client, startTime time.Time, o *LookupEventsOptions) error {
+
+		lookupOutput, err := getWriteEvents(startTime, &client)
+		if err != nil {
+			return err
+		}
+		filteredEvents, err := filterUsers(lookupOutput, Ignore, o.printAllEvents)
+		if err != nil {
+			return err
+		}
+
+		printEvents(*filteredEvents, o.printEventUrl, o.printRawEvents)
+		fmt.Println("")
+		return err
+	}
+
+	arn, accountId, err := whoami(*sts.NewFromConfig(cfg))
+	fmt.Printf("[INFO] Checking write event history since %v for AWS Account %v as %v \n", startTime, accountId, arn)
+	cloudTrailclient := cloudtrail.NewFromConfig(cfg)
+	defaultCloudtrailClient := cloudtrail.New(cloudtrail.Options{
```

Comment:
Let's keep this variable scoped to the minimum necessary. We only need it in case:  `if defaultConfig.Region != cfg.Region {`, so we can move it to that block.